src/Libraries/Microsoft.Extensions.AI.Ollama/OllamaChatClient.cs

```diff
@@ -114,12 +115,14 @@ public async IAsyncEnumerable<StreamingChatCompletionUpdate> CompleteStreamingAs
 #endif
             .ConfigureAwait(false);
 
-        await foreach (OllamaChatResponse? chunk in JsonSerializer.DeserializeAsyncEnumerable(
-            httpResponseStream,
-            JsonContext.Default.OllamaChatResponse,
-            topLevelValues: true,
-            cancellationToken).ConfigureAwait(false))
+        using var streamReader = new StreamReader(httpResponseStream);
+#if NET
+        while ((await streamReader.ReadLineAsync(cancellationToken).ConfigureAwait(false)) is { } line)
```

Comment:
FWIW this slightly changes the behavior in that it requires newline separation between single-line objects, whereas the new DeserializeAsyncEnumerable method accepts arbitrary whitespace separation between objects that could themselves be indented. But I don't think there's much we can do to improve things without STJ support.

Reply:
Yes, Ollama uses newlines.